c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,117 @@
-﻿using Newtonsoft.Json;
+﻿using JsonDiffPatchDotNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 using System;
 using System.Collections;
 using System.Collections.Generic;
 
 namespace c_sharp_json_diff
 {
-    public class Result {
-        public List<Diff> Diffs { get; set; } = new List<Diff>();
-
-    }
-    public class Diff
-    {
-        [JsonProperty("property")]
-        public string jsonProperty { get; set; } = "";
-        [JsonProperty("previous")]
-        public double previous { get; set; }
-        [JsonProperty("current")]
-        public double current { get; set; }
-        [JsonProperty("difference")]
-        public double difference { get; set; }
-    }
     public class JsonDiff
     {
-        public static Result GetDiff(IDictionary<string, double> first, IDictionary<string, double> second)
+        private bool _isAbsoluteDiff;
+        private bool _isDiffIndented;
+
+        public JsonDiff(bool isAbsoluteDiff, bool isDiffIndented)
+        {
+            _isAbsoluteDiff = isAbsoluteDiff;
+            _isDiffIndented = isDiffIndented;
+        }
+
+        public JsonDiff()
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
+            _isAbsoluteDiff = true;
+            _isDiffIndented = true;
+        }
+
+        public JObject NumericDiff(string left, string right)
+        {
+            string diff = "";
+            try
             {
-                if (firstKeys.Contains(key))
+                diff = JsonDiffBetweenTwo(left, right);
+                if (diff == null)
                 {
-                    matches.Add(key);
+                    return JObject.Parse(@"{}");
                 }
+                JObject diffJsonObject = JObject.Parse(diff);
+                JObject diffReturnObject = CalculateNumericDiff(diffJsonObject, _isAbsoluteDiff);
+                return diffReturnObject;
             }
-            foreach(string key in matches)
+            catch (Exception e)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                return JObject.Parse(diff + e.StackTrace);
```

Comment:
Instead of swallowing the exception, wrap it in a custom exception type and throw that (maybe add some docs on the method to state it can thrown an exception).

That way the client code can just do a try-catch on that exception type to see if the conversion failed, instead of inspecting the return object.